packages/shared/lib/sdk/sync.ts

```diff
@@ -59,6 +63,33 @@ interface DataResponse {
     [index: string]: unknown | undefined | string | number | boolean | Record<string, string | boolean | number | unknown>;
 }
 
+export enum PaginationType {
+    CURSOR = 'cursor',
+    LINK = 'link',
+    OFFSET = 'offset'
+}
+
+interface Pagination {
+    type: string;
+    limit?: number;
+    response_data_path?: string;
+    limit_parameter_name: string;
+}
+
+export interface CursorPagination extends Pagination {
```

Comment:
The tricky thing to understand here IMO is that one is for the response, the other for the request. 

Just a suggestion, but we could have:
- `cursor_path_in_response`
- `cursor_name_in_request`

I would also imply the `next_` here. 

In any case, I would brainstorm about the best names here because they will likely stay a while (unless we break the interface).

Reply:
In my current scheme, I use `path` for something that should be fetched from the response and `name` for something that is set on the request later on. I agree that `in_request`/`in_response` are a bit more explicit about the way these parameters are going to be used so I like this approach.

> I would also imply the next_ here.

👍

Reply:
@khaliqgant curious to have your opinion on naming!